server/odc-service/src/main/java/com/oceanbase/odc/service/diagnose/fulllinktrace/JaegerConverter.java

```diff
@@ -0,0 +1,150 @@
+    private static final String PROCESS_ID_PATTERN = "%s-%s-%s";
+
+    @Override
+    public String convert(TraceSpan span) {
+        Map<String, Object> dataMap = new HashMap<>();
+        dataMap.put("traceID", span.getLogTraceId());
+        List<Span> spans = new ArrayList<>();
+        Map<String, Process> processes = new HashMap<>();
+        dataMap.put("spans", spans);
+        dataMap.put("processes", processes);
+
+        dfs(spans, processes, span, null);
+
+        Map<String, Object> root = new HashMap<>();
+        root.put("data", Collections.singletonList(dataMap));
+        return new GsonBuilder().setPrettyPrinting().create().toJson(root);
+    }
+
+    private void dfs(List<Span> spans, Map<String, Process> processes, TraceSpan spanNode, TraceSpan parent) {
+        if (spanNode == null) {
+            return;
+        }
+
+        String processId =
+                String.format(PROCESS_ID_PATTERN, spanNode.getNode(), spanNode.getHost(), spanNode.getPort());
+        processes.putIfAbsent(processId,
+                new Process(spanNode.getNode().name(),
+                        Arrays.asList(new Tag("rec_svr_ip", spanNode.getHost()),
+                                new Tag("rec_svr_port", spanNode.getPort() + ""),
+                                new Tag("tenant_id", spanNode.getTenantId() + ""))));
+
+        spans.add(Span.builder()
+                .traceID(spanNode.getTraceId())
+                .spanID(spanNode.getSpanId())
+                .operationName(spanNode.getSpanName())
+                .startTime(toTimestampWithMicros(spanNode.getStartTimestamp()))
+                .duration(spanNode.getElapseMicroSeconds())
+                .references(parent == null ? null
+                        : Collections.singletonList(new Reference(parent.getTraceId(), parent.getSpanId())))
+                .tags(CollectionUtils.isEmpty(spanNode.getTags()) ? null
+                        : spanNode.getTags()
+                                .stream().map(map -> {
+                                    Entry<String, Object> entry = map.entrySet().iterator().next();
+                                    return new Tag(entry.getKey(), entry.getValue().toString());
+                                }).collect(Collectors.toList()))
+                .processID(processId)
+                .build());
+        if (CollectionUtils.isEmpty(spanNode.getSubSpans())) {
+            return;
+        }
+        for (TraceSpan child : spanNode.getSubSpans()) {
+            dfs(spans, processes, child, spanNode);
+        }
+    }
+
+    private long toTimestampWithMicros(String timeStr) {
+        Instant instant = LocalDateTime.parse(timeStr, TraceSpan.TIMESTAMP_FORMATTER)
+                .atZone(ZoneId.systemDefault()).toInstant();
+        return instant.getEpochSecond() * 1000000 + instant.getNano() / 1000;
+    }
+
+    @Getter
+    @Builder
+    private static class Span {
+        private String traceID;
```

Comment:
here 'traceID', 'spanID', 'processID' use 'ID' postfix, other module use 'Id' instead.

what is the reason of diferent style ?

Reply:
because Jaeger can only recognize this format rather than `Id` 😂